library/core/class.upload.php

```diff
@@ -168,6 +168,10 @@ public static function parse($Name) {
         $Name = str_replace('\\', '/', $Name);
         $PathUploads = str_replace('\\', '/', PATH_UPLOADS);
 
+        if (c('Garden.AllowSSL') || c('Garden.ForceSSL')) {
+            $Name = preg_replace("/^http:/i", "https:", $Name);
+        }
+
         if (preg_match('`^https?://`', $Name)) {
```

Comment:
You already have a pregmatch here. You should not preg_replace for every call to parse.